README.md

```diff
@@ -68,7 +68,7 @@ new ExtractTextPlugin(options: filename | object)
 |Name|Type|Description|
 |:--:|:--:|:----------|
 |**`id`**|`{String}`|Unique ident for this plugin instance. (For advanced usage only, by default automatically generated)|
-|**`filename`**|`{String}`|Name of the result file. May contain `[name]`, `[id]` and `[contenthash]`|
+|**`filename`**|`{String|Object}`|Name of the result file. May contain `[name]`, `[id]` and `[contenthash]`|
```

Comment:
I see. It's that `compilation.getPath` which is the problem. Basically `modify` triggers **after** that.

The design could be inverted like this:

```javascript
filename: (getPath, filename) => getPath(filename)
```

That `getPath` would run `compilation.getPath` with all its parameters and provide the hook that would give you maximum amount of control. Now you could manipulate `filename` before and after placeholders get applied while keeping the API surface minimal.